### PR TITLE
Adds support for mouse events.

### PR DESCRIPTION
**Pull request by DracoNoir**
Addresses issue #100.

### CODE REVIEW

**Comment by DracoNoir** [comment]
This also addresses issue #89.

File: google-map.html [hunk 1]
```diff
@@ -173,6 +228,21 @@
...
+        this._forwardEvent('mousemove');
+        this._forwardEvent('mouseout');
+        this._forwardEvent('mouseover');
+        this._forwardEvent('mouseup');
+        this._forwardEvent('rightclick');
+      },
+
+      _forwardEvent: function(name) {
+        google.maps.event.addListener(this.marker, name, function(event) {
+          this.fire('google-map-marker-' + name, event);
```
Comment (positlabs) [inline google-map.html]: Why not just fire the name of the event? Seems like it would be cleaner that way.

`this.fire(name, event);`

Comment (DracoNoir) [inline google-map.html]: I chose the more verbose form to allow a developer to register handlers above the element (e.g. on the google-map itself) and still disambiguate the source by type.  However, there is also value in brevity (and the handler will be passed the source element).  Do you have a recommendation @ebidel?

Comment (ebidel) [inline google-map.html]: We've encouraged the use of `ELEMENT-NAME-event` in the style guide https://github.com/GoogleWebComponents/style-guide#events

**Comment by DracoNoir** [comment]
Thanks for the the suggestion @ebidel.  I updated the descriptions to use your wording.

File: google-map.html [hunk 2]
```diff
@@ -34,7 +34,62 @@
...
+
+<!--
+Fired for a mouseup on the marker.
+
+@event google-map-marker-mouseup
+@param {google.maps.MouseEvent} event The mouse event.
+-->
+
+<!--
+Fired for a rightclick on the marker.
```
Comment (ebidel) [inline google-map.html]: Can you also add to these descriptions that these events are only fired if clickEvents/mouseEvents are set and refer people to those properties.

Comment (DracoNoir) [inline google-map.html]: Great idea.  Done.

**Comment by ebidel** [comment]
:+1:

**Pull request merged**